app/javascript/components/conditions/ConditionSetFormField/model.js

```diff
@@ -71,7 +71,10 @@ class ConditionSetModel {
   mapConditionsToStores(conditions) {
     // Only modify if necessary to prevent a cycle.
     if (conditions.some((condition) => !(condition instanceof ConditionModel))) {
-      return conditions.map((condition) => new ConditionModel(condition));
+      return conditions.map((condition) => {
+        condition.refableQings = this.refableQings;
```

Comment:
`mapConditionsToStores` is here to make sure all conditions are instances of `ConditionModel` and not just plain objects -- this code won't be invoked if the condition starts out as a `ConditionModel`.

I suggest separating concerns and making `refableQings` an expected param in the `ConditionModel` constructor. Then do e.g.

```js
conditions.map((condition) => new ConditionModel({ ...condition, refableQings: this.refableQings });
```

Reply:
Good catch. I went with something similar to what you suggested. Hopefully you like it :)